core/src/main/java/org/elasticsearch/transport/ConnectionProfile.java

```diff
@@ -79,6 +79,10 @@ public void addConnections(int numConnections, TransportRequestOptions.Type... t
             offset += numConnections;
         }
 
+        /**
+         * Creates a new {@link ConnectionProfile} based on the added connections.
+         * @throws IllegalArgumentException if any of the {@link org.elasticsearch.transport.TransportRequestOptions.Type} enum is missing
```

Comment:
This Javadoc is incorrect; the method throws an `IllegalStateException` (which in this case I think is correct). :smile: